src/components/pages/career-details/new-career-form/career-form.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+import { SIMPLE_EMAIL_PATTERN } from '../../../forms/constants';
+import { FileDropperType } from '../../../forms/file-dropper/file-dropper';
+import { TextArea } from '../../../forms/text-area/text-area';
+import { TextInput } from '../../../forms/text-input/text-input';
+import { FormLayout } from '../../contact/form';
+import { SectionHeadline } from '../job-description';
+import {
+  CareerDetailsCheckbox,
+  CareerDetailsError,
+  CareerDetailsFileText,
+  CareerDetailsFileUpload,
+  CareerDetailsSubmitButton,
+  CareerDetailsSuccess,
+} from './career-form-fields';
+import { uploadToPersonio } from './upload';
+
+interface CareerFormProps {
+  company_id: string;
+  access_token: string;
+  job_position_id: string;
+  recruiting_channel_id: string;
+  scrollToStart: () => void;
+}
+
+export type FormDataProps = {
+  first_name: string;
+  last_name: string;
+  email: string;
+  documents: FileDropperType[];
+  message: string;
+  phone?: string;
+  location: string;
+  available_from: string;
+  salary_expectations: string;
+
+  privacy: boolean;
+};
+
+export type FormErrors = { api?: never };
+
+export const Form = (props: CareerFormProps) => {
+  const {
+    setValue,
+    setError,
+    clearErrors,
+    register,
+    handleSubmit,
+    control,
+    watch,
+    formState: { errors, isSubmitSuccessful, isSubmitting },
+  } = useForm<FormDataProps & FormErrors>({
+    mode: 'onSubmit',
+  });
+
+  const { t } = useTranslation();
+  const selectedFiles = watch('documents');
+
+  const onValidateForm = () => {
+    // to allow re send after api error
+    // there should not be an old api error when pressing submit
+    if (errors.api) {
+      clearErrors('api');
+    }
+
+    // since documents is not controlled, it has to be validated manually
+    if (selectedFiles?.length === 0 || !selectedFiles) {
+      setError(
+        'documents',
+        { type: 'manual', message: t<string>('career.error.cv') },
+        { shouldFocus: true },
+      );
+    }
+
+    for (let i = 0; i < selectedFiles?.length; i++) {
+      if (!selectedFiles[i].fileCategory)
+        setError('documents', {
+          type: 'manual',
+          message: t<string>('career.error.category'),
+        });
+    }
+  };
```

Comment:
This validation of the selectedFiles is generally something i am not happy with. I am already thinking of a way to put this whole thing into the file dropper.

But for now, I would suggest to leave it like this, and I will plan to refactor the file dropper some day to a controlled component, I have an idea in which way it could work, but it is not implementation ready